curator-framework/src/main/java/org/apache/curator/framework/state/ConnectionStateListener.java

```diff
@@ -31,14 +31,14 @@
     void stateChanged(CuratorFramework client, ConnectionState newState);
 
     /**
-     * Normally, ConnectionStateListeners are decorated via the configured
-     * {@link org.apache.curator.framework.state.ConnectionStateListenerDecorator}. For certain
-     * critical cases, however, this is not desired. If your listener returns <code>true</code>
-     * for doNotDecorate(), it will not be passed through the decorator.
+     * ConnectionStateListener managers set via {@link org.apache.curator.framework.CuratorFrameworkFactory.Builder#connectionStateListenerManagerFactory(ConnectionStateListenerManagerFactory)}
+     * are allowed to proxy (etc.) ConnectionStateListeners as needed. If this method returns <code>true</code>
+     * the ConnectionStateListener manager must <em>not</em> proxy the listener as it's a vital internal
+     * listener used by Curator.
      *
      * @return true/false
      */
-    default boolean doNotDecorate()
+    default boolean doNotProxy()
```

Comment:
I think proxy is a bit vague. Why not something like useCircuitBreaker() instead of doNotProxy(), to be more clear about the context it is used in? Unless you see future possibilities to call it also for other 'proxies' than the circuit breaker.

Reply:
It’s not specific to circuit breaking.

Reply:
Gotcha